File: adjacentElementsProduct/adjacent_elements_product.py

from typing import List


def adjacent_elements_product(input_array: List[int]) -> int:
    products = [
        n * input_array[idx+1]
        for idx, n in enumerate(input_array) 
        if 0 <= (idx+1) < len(input_array)
    ]

    return max(products)


inputs = [
    [3, 6, -2, -5, 7, 3],
    [3, 2, 9, 7, 4, 5],
    [-1, -5, -9, -3, 4, 7]
]

for ia in inputs:
    print(f'{ia} -> {adjacent_elements_product(ia)}')

File: adjacentElementsProduct/adjacentElementsProduct.ts
const adjacentElementsProduct = (inputArray: number[]): number => {
	const products = inputArray
		.map( (x, i) => inputArray[i+1] 
			? x * inputArray[i+1] 
			: false )
		.filter(x=>x)

	return Math.max.apply(null, products);
}


console.time('adjacentElementsProduct x3');

console.log(adjacentElementsProduct([3, 6, -2, -5, 7, 3]));
console.log(adjacentElementsProduct([3, 2, 9, 7, 4, 5]));
console.log(adjacentElementsProduct([-1, -5, -9, -3, 4, 7]));

console.timeEnd('adjacentElementsProduct x3');
